### PR TITLE
Add valid_thresh to contrib.box_nms

### PR DESCRIPTION
**Pull request by ijkguo**
## Description ##
Add `valid_thresh`, a filter of confidence score to box_nms to reduce computation. It can be combined with the existing `topk` filter. @zhreshold 
Some results from a pre-trained faster rcnn model:

| description | valid_thresh | topk | mAP (%) | box_nms cpu time (ms) | box_nms gpu time (ms) |
|---|---|---|---|---|---|
| original | 0 | -1 | 74.17 | 4.7 | 17 |
| topk only | 0 | 100 | 73.93 | 2.1 | 16 |
| thresh only | 1e-3 | -1 | 74.17 | 2.6 | 3 |
| topk + thresh | 1e-3 | 100 | 73.90 | 0.5 | 2 |

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/operator/contrib/bounding_box-inl.h [hunk 1]
```diff
@@ -145,6 +152,60 @@ inline uint32_t BoxNMSNumVisibleOutputs(const NodeAttrs& attrs) {
...
+    if (scores[i] > valid_thresh) {
+      out_scores[j] = scores[i];
+      out_sorted_index[j] = sorted_index[i];
+      j++;
+    }
+  }
+  return j;
+}
+
+#ifdef __CUDACC__
```
Comment (zhreshold) [inline src/operator/contrib/bounding_box-inl.h]: move to bounding_box-inl.cuh and include

Comment (zhreshold) [reply src/operator/contrib/bounding_box-inl.h]: and the thrust headers as well

**Comment by zhreshold** [comment]
Essentially looks good, please see comments
